File: server/app/src/main/java/ee/rsx/kata/bank/app/rest/loans/validation/ValidationEndpoints.java
package ee.rsx.kata.bank.app.rest.loans.validation;

import ee.rsx.kata.bank.loans.validation.limits.LoadValidationLimits;
import ee.rsx.kata.bank.loans.validation.limits.ValidationLimitsDTO;
import ee.rsx.kata.bank.loans.validation.ssn.SsnValidationResultDTO;
import ee.rsx.kata.bank.loans.validation.ssn.ValidateSocialSecurityNumber;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/loans/validation")
@RequiredArgsConstructor
public class ValidationEndpoints {

  private final LoadValidationLimits loadValidationLimits;
  private final ValidateSocialSecurityNumber validateSocialSecurityNumber;

  @GetMapping(value = "/limits")
  public ValidationLimitsDTO loadValidationLimits() {
    return loadValidationLimits.invoke();
  }

  @GetMapping(value = "/ssn")
  public SsnValidationResultDTO validateSsn(@RequestParam("value") String value) {
    return validateSocialSecurityNumber.on(value);
  }
}

File: server/app/src/main/kotlin/ee/rsx/kata/bank/app/rest/loans/validation/ValidationEndpoints.kt
package ee.rsx.kata.bank.app.rest.loans.validation

import ee.rsx.kata.bank.loans.validation.limits.LoadValidationLimits
import ee.rsx.kata.bank.loans.validation.ssn.ValidateSocialSecurityNumber
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/loans/validation")
class ValidationEndpoints(
  private val loadLimits: LoadValidationLimits,
  private val validateSocialSecurityNumber: ValidateSocialSecurityNumber
) {

  @GetMapping(value = ["/limits"])
  fun loadValidationLimits() = loadLimits()

  @GetMapping(value = ["/ssn"])
  fun validateSsn(@RequestParam("value") value: String) = validateSocialSecurityNumber(value)
}
